utils/python/CIME/XML/component.py

```diff
@@ -43,11 +40,12 @@ def get_valid_model_components(self):
         components = comps.split(',')
         return components
 
-    def _get_value_match(self, node, attributes=None):
+    def _get_value_match(self, node, attributes=None, exact_match=False):
```

Comment:
Matching for components and compsets is regex matching, while matching for namelists is exact.